packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieveSourcePath.ts

```diff
@@ -40,75 +36,60 @@ export class LibraryRetrieveSourcePathExecutor extends RetrieveExecutor<
-          sourcePath
-        );
-        if (isInSfdxPackageDirectory) {
-          return inputs;
-        }
-      } catch (error) {
-        telemetryService.sendException(
-          'force_source_retrieve_with_sourcepath',
-          `Error while parsing package directories. ${error.message}`
-        );
-      }
-
-      const errorMessage = nls.localize(
-        'error_source_path_not_in_package_directory_text'
-      );
-      telemetryService.sendException(
-        'force_source_retrieve_with_sourcepath',
-        errorMessage
-      );
-      notificationService.showErrorMessage(errorMessage);
-      channelService.appendLine(errorMessage);
-      channelService.showChannelOutput();
-    }
-    return { type: 'CANCEL' };
+export const forceSourceRetrieveSourcePaths = async (
+  sourceUri: vscode.Uri,
+  uris: vscode.Uri[] | undefined
+) => {
+  // When a single file is selected and "Retrieve Source from Org" is executed,
+  // sourceUri is passed, and the uris array contains a single element, the same
+  // path as sourceUri.
+  //
+  // When multiple files are selected and "Retrieve Source from Org" is executed,
+  // sourceUri is passed, and is the path to the first selected file, and the uris
+  // array contains an array of all paths that were selected.
+  //
+  // When editing a file and "Retrieve This Source from Org" is executed,
+  // sourceUri is passed, but uris is undefined.
+  if (!uris || uris.length < 1) {
+    uris = [];
+    uris.push(sourceUri);
   }
-}
 
-export async function forceSourceRetrieveSourcePath(explorerPath: vscode.Uri) {
-  if (!explorerPath) {
-    const editor = vscode.window.activeTextEditor;
-    if (editor && editor.document.languageId !== 'forcesourcemanifest') {
-      explorerPath = editor.document.uri;
-    } else {
-      const errorMessage = nls.localize(
-        'force_source_retrieve_select_file_or_directory'
-      );
-      telemetryService.sendException(
-        'force_source_retrieve_with_sourcepath',
-        errorMessage
-      );
-      notificationService.showErrorMessage(errorMessage);
-      channelService.appendLine(errorMessage);
-      channelService.showChannelOutput();
-      return;
+  const messages: ConflictDetectionMessages = {
+    warningMessageKey: 'conflict_detect_conflicts_during_retrieve',
+    commandHint: inputs => {
+      const commands: string[] = [];
+      (inputs as string[]).forEach(input => {
+        commands.push(
+          new SfdxCommandBuilder()
+            .withArg('force:source:retrieve')
+            .withFlag('--sourcepath', input)
+            .build()
+            .toString()
+        );
+      });
+      const hints = commands.join('\n  ');
+
+      return hints;
     }
-  }
+  };
 
   const commandlet = new SfdxCommandlet(
     new SfdxWorkspaceChecker(),
-    new FilePathGatherer(explorerPath),
+    new LibraryPathsGatherer(uris),
     new LibraryRetrieveSourcePathExecutor(),
-    new SourcePathChecker()
```

Comment:
@jag-sfdc I removed SourcePathChecker() and replaced with TimestampConflictChecker().  This seems to work but want to make you aware of this change.

Reply:
@jeffb-sfdc - At one time we had both SourcePathChecker() and TimeStampConflictChecker() as in this PR- https://github.com/forcedotcom/salesforcedx-vscode/pull/3348/files. We then removed TimeStampConflictChecker likely to fix a bug. I'll dig through the PRs to understand why we did this. If this isn't a required fix for multi file deploy/retrieve to work, can we not change it now.

Reply:
It's not a requirement.  I'll add it back in.  I removed it b/c TimestampConflictChecker() worked for both deploy & retrieved, and thought I'd consolidate.

Reply:
@jeffb-sfdc - I was able to track down the reason. We decided to disable conflict detection for retrieves and pushed a PR for it - https://github.com/forcedotcom/salesforcedx-vscode/pull/3382. We should leave this with SourcePathChecker() for now.

Reply:
Thanks, I reverted it back.